course_catalog/etl/youtube.py

```diff
@@ -144,21 +148,26 @@ def extract():
         A generator that yields tuples with offered_by and video data
 
     """
-    if not (settings.YOUTUBE_CONFIG_FILE_LOCATION and settings.YOUTUBE_DEVELOPER_KEY):
+    if not settings.YOUTUBE_DEVELOPER_KEY:
         return
 
     youtube_client = get_youtube_client()
-    response = requests.get(settings.YOUTUBE_CONFIG_FILE_LOCATION)
-    response.raise_for_status()
-
-    channels_yml = yaml.safe_load(response.content)
 
-    if channels_yml and ("channels" in channels_yml):
-        for channel in channels_yml["channels"]:
-            if "playlists" in channel:
-                for playlist_id in channel["playlists"]:
+    branch_name = settings.OPEN_VIDEO_DATA_BRANCH or "master"
+    github_client = github.Github()
+    repo = github_client.get_repo(CONFIG_FILE_REPO)
+
+    for file in repo.get_contents(CONFIG_FILE_FOLDER, ref=branch_name):
+        channel_data = yaml.safe_load(file.decoded_content)
+        if (
+            channel_data
+            and ("playlists" in channel_data)
+            and ("offered_by" in channel_data)
+        ):
+            for playlist in channel_data["playlists"]:
+                if "id" in playlist:
```

Comment:
Should also check that `isinstance(playlist, dict)` since `"id" in "j4G8fvmidm4"i` is `True`